src/module/implements/implementBenefits/tome.js

```diff
@@ -404,3 +440,20 @@ Hooks.on("deleteImplementEffects", (a) => {
 
   deleteOldTomeEffect(oldTome);
 });
+
+Hooks.on("renderChatMessage", (message) => {
+  if (
+    !message.flags.pf2e.context?.options.some(
+      (o) => o === "target:mark:tome-adept-rk-success"
+    ) ||
+    !game.ready ||
+    !message.actor.items.some(
+      (i) => i.sourceId === TOME_ADEPT_RK_EFFECT_UUID
+    ) ||
+    message.actor.primaryUpdater !== game.user
+  )
+    return;
+  message.actor.items
+    .find((i) => i.sourceId === TOME_ADEPT_RK_EFFECT_UUID)
+    .delete();
+});
```

Comment:
I asked on discord, sounds like primaryUpdater is the right way to only trigger on one user's client and not everyone who owns the actor all trying to delete the effect.  

I'd check `!game.ready` first.  The expressions in an `||` get done in order until one is true.  So as it is now, it will scan the all the options of every single message in the chat history on connect.  Then ignore all the messages anyway because game.ready isn't true.

Then I'd check primary updater, since it's quicker to check than scanning the items.

I think maybe it should be `message.flags.pf2e?.context?.options.some()`  I think all message have flags, but they might not have pf2e flags.  Not sure about that.

Using `actor.itemsTypes.effect.xxx` should be faster, since it only has to scan the effects.

And it's not really necessary to check for the effect twice.  Just change the last line to:
`message.actor.itemTypes.effect.find(....)?.delete();`

It'll delete if it exists and do nothing if it doesn't.

Reply:
That's all just suggestions.  I think it totally works as is.

Reply:
Thanks for looking over it!